_posts/2016-05-11-aws-lambdas.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+name: aws-lambdas 
+title: AWS Lambda for Beginners
+date: 2016-05-11 00:20:00 +00:00
+author: Mashooq Badar 
+image:
+   src: /assets/img/custom/blog/aws-lambda.png
+tags:
+- microservices 
+- devops 
+- aws
+---
+
+AWS Lambda is a compute service from Amazon. It makes deployment and provisioning very simple and fits very well with microservices based architecture. You can find out more about AWS Lambda [here](http://docs.aws.amazon.com/lambda/latest/dg/welcome.html). Currently supported platforms are JVM, Node JS and Python. 
+
+The programming model for AWS Lambda consists of **Handler, Context Object, Logging and Exceptions**. These are described [here](http://docs.aws.amazon.com/lambda/latest/dg/programming-model-v2.html). An instance of AWS Lambda must not hold state because it will be stopped, started and replicated as needed. Persistent state should be stored in a service that is outside the lifecycle of the lambda such as Amazon DynamoDB, S3 etc.
+
+First of all follow the instructions [here](http://docs.aws.amazon.com/lambda/latest/dg/setup.html) to setup an AWS Account and AWS Command-line Interface and note down your account id. 
+
+### Step 1: The Code
+The most basic lambda will look like the following in Python:
+
+```python
+def lambda_handler(event, context):
+  return "Hello World!"
+```
+
+or like the following in Java:
+
+```java
+package example;
+
+import com.amazonaws.services.lambda.runtime.Context; 
+
+public class Hello {
+    public String lambdaHandler(String event, Context context) {
+        return "Hello World!";
+    }
+}
+```
+
+
+You can follow the, somewhat lengthy, instructions [here](http://docs.aws.amazon.com/lambda/latest/dg/get-started-create-function.html) to deploy this function … but that’s no fun! Let’s do it devops style ;)
+
+Paste the above Python code in a file called ```helloworld.py```. If you want to use the Java version then follow the instructions [here](http://docs.aws.amazon.com/lambda/latest/dg/java-create-jar-pkg-maven-no-ide.html) to build your lambda and create a deployment package using Maven.
+
+### Step 2: The Role
+Create a ```trust.json``` file . The trust allows our function to assume the [role](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) of an AWS Lambda. 
+
+In ```trust.json``` we are allowing the function to assume the role of a ```lambda.amazonaws.com``` service.
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [{
+    "Sid": "",
+    "Effect": "Allow",
+    "Principal": {
+      "Service": "lambda.amazonaws.com"
+    },
+    "Action": "sts:AssumeRole"
+  }]
+}
+```
+
+### Step 3: The Deployment 
```

Comment:
The result of the deployment is the execution detailed in the next step. What better way to demonstrate the result of the deployment then by actually using the deployed behaviour?